### PR TITLE
Fix batch_dot with axis=None for 2D tensors #9847

### PR DESCRIPTION
**Pull request by vkk800**
See this issue: https://github.com/keras-team/keras/issues/9847

I created a quick fix that, for 2D tensors, defaults to doing the dot product on the axis index 1 if no axes arguments is provided. From the documentation I gather that this is logical considering how batch_dot() is typically used. Now batch_dotting two 2D-tensors with same shapes results, by default, in a new tensor with the same batch dimension, but the second dimension reduced to 1.

The example case given in the issue still throws error (because in that case the dimensions of the tensors are not compatible with the default behavior I provided), but the error user gets is now more informative.

### CODE REVIEW

**Comment by fchollet** [comment]
Please add a unit test.

**Comment by vkk800** [comment]
While making the unit tests, I discovered that Theano and CNTK backends apparently default to different behavior when called without axes argument:
```
if axes is None:
    axes = [len(x_shape) - 1, len(y_shape) - 2]
```

I'm not sure what is the logic behind this; the Theano & CNTK behavior means that batch_dot() fails when called for example for two 2D tensors of same non-symmetric shapes. I think the default behavior that I defined for tensorflow was more natural, but of course they should all behave consistently. Any opinions on which behavior we should choose?

**Comment by vkk800** [comment]
Okay, I had time today to look further into this and it seems that Theano and TensorFlow backends give completely inconsistent results when using this batch_dot(). I ran a simple set of tests I wrote using both Theano and TensorFlow backends and get very different results in many cases. The most obvious thing is that Theano throws an error if you try to sum over this 0th dimension of the matrix (this comes from Theano implementation of the function), while TensorFlow doesn't. What is more alarming is that the values and even the dimensions of the outputs are different in many cases.

In case someone wants to look at my tests, here are the tests: https://pastebin.com/Ka7Z03nU and here are the Theano results: https://pastebin.com/0JTneTpU and here are the TensorFlow results: https://pastebin.com/ec4kNpjp . If the two backends gave same results, all the lines in the results would match.

I will look into it, but I can't promise any quick results, because I'm not familiar with the backend stuff so much. I think that maybe I will make my test code a bit easier to read and then I'll make an issue about this. This pull request should probably be closed since fixing just this trivial issue in batch_fix is pointless if it otherwise doesn't work as intended.
